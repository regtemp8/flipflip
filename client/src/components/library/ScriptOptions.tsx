import {
  Button,
  Collapse,
  Dialog,
  DialogActions,
  DialogContent,
  Divider,
  Grid,
  type Theme,
  Typography
} from '@mui/material'

import createStyles from '@mui/styles/createStyles'
import withStyles, { type WithStyles } from '@mui/styles/withStyles'

import { green, red } from '@mui/material/colors'

import FontOptions from './FontOptions'
import BaseSwitch from '../common/BaseSwitch'
import {
  setCaptionScriptStopAtEnd,
  setCaptionScriptNextSceneAtEnd,
  setCaptionScriptSyncWithAudio,
  setCaptionScriptOpacity
} from '../../store/captionScript/actions'
import { setCaptionScript } from '../../store/captionScript/slice'
import {
  selectCaptionScript,
  selectCaptionScriptStopAtEnd,
  selectCaptionScriptNextSceneAtEnd,
  selectCaptionScriptSyncWithAudio,
  selectCaptionScriptOpacity
} from '../../store/captionScript/selectors'
import BaseSlider from '../common/slider/BaseSlider'
import { useAppDispatch, useAppSelector } from '../../store/hooks'

const styles = (theme: Theme) =>
  createStyles({
    bpmProgress: {
      position: 'absolute',
      right: 67
    },
    tagProgress: {
      position: 'absolute',
      right: 20
    },
    success: {
      backgroundColor: green[500],
      '&:hover': {
        backgroundColor: green[700]
      }
    },
    failure: {
      backgroundColor: red[500],
      '&:hover': {
        backgroundColor: red[700]
      }
    },
    actions: {
      marginRight: theme.spacing(3)
    },
    fullWidth: {
      width: '100%'
    },
    noPadding: {
      padding: '0 !important'
    },
    endInput: {
      paddingLeft: theme.spacing(1),
      paddingTop: 0
    },
    percentInput: {
      minWidth: theme.spacing(11)
    },
    toggleFont: {
      marginLeft: 'auto'
    },
    fontDivider: {
      marginTop: theme.spacing(1),
      marginBottom: theme.spacing(2)
    },
    fontProgress: {
      position: 'absolute'
    }
  })

export interface ScriptOptionsProps extends WithStyles<typeof styles> {
  scriptID: number
  onDone: () => void
}

function ScriptOptions(props: ScriptOptionsProps) {
  const dispatch = useAppDispatch()
  const originalScript = useAppSelector(
    selectCaptionScript(props.scriptID),
    () => true
  )
  const stopAtEnd = useAppSelector(selectCaptionScriptStopAtEnd(props.scriptID))
  const nextSceneAtEnd = useAppSelector(
    selectCaptionScriptNextSceneAtEnd(props.scriptID)
  )

  const onCancel = () => {
    dispatch(setCaptionScript(originalScript))
    props.onDone()
  }

  const classes = props.classes
  return (
    <Dialog open={true} onClose={onCancel} aria-describedby="edit-description">
      <DialogContent>
        <Typography variant="h6">Edit script options</Typography>
        <Grid container spacing={2} alignItems="center">
          <Grid item xs={12}>
            <Grid container spacing={2} alignItems="center">
              <Grid item>
                <Collapse in={!nextSceneAtEnd}>
                  <BaseSwitch
                    label="Stop at End"
                    size="small"
                    selector={selectCaptionScriptStopAtEnd(props.scriptID)}
                    action={setCaptionScriptStopAtEnd(props.scriptID)}
                  />
                </Collapse>
                <Collapse in={!stopAtEnd}>
                  <BaseSwitch
                    label="Next Scene at End"
                    size="small"
                    selector={selectCaptionScriptNextSceneAtEnd(props.scriptID)}
                    action={setCaptionScriptNextSceneAtEnd(props.scriptID)}
                  />
                </Collapse>
                <BaseSwitch
                  label="Sync Timestamp with Audio"
                  size="small"
                  selector={selectCaptionScriptSyncWithAudio(props.scriptID)}
                  action={setCaptionScriptSyncWithAudio(props.scriptID)}
                />
              </Grid>
            </Grid>
          </Grid>
          <Grid item xs={12}>
            <BaseSlider
              min={0}
              max={100}
              selector={selectCaptionScriptOpacity(props.scriptID)}
              action={setCaptionScriptOpacity(props.scriptID)}
              labelledBy="opacity-slider"
              label={{ text: 'Script Opacity:', appendValue: true }}
              format={{ type: 'percent' }}
            />
          </Grid>
          <Grid item xs={12}>
            <FontOptions
              name={'Blink'}
              captionScriptId={props.scriptID}
              type="blink"
            />
            <Divider className={classes.fontDivider} />
            <FontOptions
              name={'Caption'}
              captionScriptId={props.scriptID}
              type="caption"
            />
            <Divider className={classes.fontDivider} />
            <FontOptions
              name={'Big Caption'}
              captionScriptId={props.scriptID}
              type="captionBig"
            />
            <Divider className={classes.fontDivider} />
            <FontOptions
              name={'Count'}
              captionScriptId={props.scriptID}
              type="count"
            />
          </Grid>
        </Grid>
      </DialogContent>
      <DialogActions className={classes.actions}>
        <Button onClick={onCancel} color="secondary">
          Cancel
        </Button>
        <Button onClick={props.onDone} color="primary">
          Save
        </Button>
      </DialogActions>
    </Dialog>
  )
}

;(ScriptOptions as any).displayName = 'ScriptOptions'
export default withStyles(styles)(ScriptOptions as any)
