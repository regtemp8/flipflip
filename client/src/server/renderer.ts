import createCache, { type EmotionCache } from '@emotion/cache'
import { configureStore } from '@reduxjs/toolkit'
import { AppStorage, SystemConstants } from 'flipflip-common'
import createAppReducer from '../store/app/slice'
import createAudioReducer from '../store/audio/slice'
import createCaptionScriptReducer from '../store/captionScript/slice'
import createClipReducer from '../store/clip/slice'
import createLibrarySourceReducer from '../store/librarySource/slice'
import createPlaylistReducer from '../store/playlist/slice'
import createSceneReducer from '../store/scene/slice'
import createSceneGroupReducer from '../store/sceneGroup/slice'
import createTagReducer from '../store/tag/slice'
import playersSlice from '../store/player/slice'
import videoClipperSlice from '../store/videoClipper/slice'
import scenePickerSlice from '../store/scenePicker/slice'
import captionScriptorSlice from '../store/captionScriptor/slice'
import sceneDetailSlice from '../store/sceneDetail/slice'
import audioLibrarySlice from '../store/audioLibrary/slice'
import sourceScraperSlice from '../store/sourceScraper/slice'
import createConstantsReducer from '../store/constants/slice'
import { fromAppStorage } from '../store/app/convert'
import createDisplayReducer from '../store/display/slice'
import createDisplayViewReducer from '../store/displayView/slice'
import createScenePlaylistItemReducer from '../store/scenePlaylistItem/slice'
import createDisplayPlaylistItemReducer from '../store/displayPlaylistItem/slice'

export function createEmotionCache(): EmotionCache {
  return createCache({ key: 'css' })
}

export function createReduxStore(
  appStorage?: AppStorage,
  constants?: SystemConstants
) {
  const state = appStorage != null ? fromAppStorage(appStorage) : undefined
  return configureStore({
    reducer: {
      app: createAppReducer(state?.app),
      audio: createAudioReducer(state?.audio),
      captionScript: createCaptionScriptReducer(state?.captionScript),
      clip: createClipReducer(state?.clip),
      librarySource: createLibrarySourceReducer(state?.librarySource),
      playlist: createPlaylistReducer(state?.playlist),
      displayPlaylistItem: createDisplayPlaylistItemReducer(
        state?.displayPlaylistItem
      ),
      scenePlaylistItem: createScenePlaylistItemReducer(
        state?.scenePlaylistItem
      ),
      scene: createSceneReducer(state?.scene),
      sceneGroup: createSceneGroupReducer(state?.sceneGroup),
      display: createDisplayReducer(state?.display),
      displayView: createDisplayViewReducer(state?.displayView),
      tag: createTagReducer(state?.tag),
      constants: createConstantsReducer(constants),
      // components
      players: playersSlice,
      videoClipper: videoClipperSlice,
      scenePicker: scenePickerSlice,
      captionScriptor: captionScriptorSlice,
      sceneDetail: sceneDetailSlice,
      audioLibrary: audioLibrarySlice,
      sourceScraper: sourceScraperSlice
    }
  })
}
