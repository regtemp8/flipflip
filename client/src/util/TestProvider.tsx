import * as React from 'react'
import { Provider } from 'react-redux'
import { createTheme } from '@mui/material'
import { StyledEngineProvider, ThemeProvider } from '@mui/material/styles'
import defaultTheme from 'flipflip-common'

export default class TestProvider extends React.Component {
  readonly props: {
    children: any
    store: any
  }

  disableTransitions = {
    defaultProps: {
      disablePortal: true,
      hideBackdrop: true,
      TransitionComponent: ({ children }) => children
    }
  }

  render() {
    const theme = createTheme({
      ...defaultTheme,
      // disable Transitions when runing tests, throws errors otherwise.
      components: {
        MuiMenu: this.disableTransitions,
        MuiPopover: this.disableTransitions,
        MuiDialog: this.disableTransitions,
        MuiModal: this.disableTransitions
      }
    } as any)
    return (
      <Provider store={this.props.store}>
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={theme}>{this.props.children}</ThemeProvider>
        </StyledEngineProvider>
      </Provider>
    )
  }
}
