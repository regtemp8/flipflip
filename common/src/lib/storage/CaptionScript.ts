import { copy } from '../utils';

import { type FontSettings } from './FontSettings';
import { type Tag } from './Tag';

export type CaptionScript = {
  id: number;
  url?: string;
  script?: string;
  marked: boolean;
  tags: Tag[];
  opacity: number;
  stopAtEnd: boolean;
  nextSceneAtEnd: boolean;
  syncWithAudio: boolean;
  blink: FontSettings;
  caption: FontSettings;
  captionBig: FontSettings;
  count: FontSettings;
};

const initialCaptionScript: CaptionScript = {
  id: 0,
  marked: false,
  tags: [],
  opacity: 100,
  stopAtEnd: false,
  nextSceneAtEnd: false,
  syncWithAudio: true,
  blink: {
    color: '#FFFFFF',
    fontSize: 20,
    fontFamily: 'Arial Black,Arial Bold,Gadget,sans-serif',
    border: false,
    borderpx: 5,
    borderColor: '#000000',
  },
  caption: {
    color: '#FFFFFF',
    fontSize: 8,
    fontFamily: 'Helvetica Neue,Helvetica,Arial,sans-serif',
    border: false,
    borderpx: 3,
    borderColor: '#000000',
  },
  captionBig: {
    color: '#FFFFFF',
    fontSize: 12,
    fontFamily: 'Arial Black,Arial Bold,Gadget,sans-serif',
    border: false,
    borderpx: 4,
    borderColor: '#000000',
  },
  count: {
    color: '#FFFFFF',
    fontSize: 20,
    fontFamily: 'Arial Black,Arial Bold,Gadget,sans-serif',
    border: false,
    borderpx: 5,
    borderColor: '#000000',
  },
};

export function newCaptionScript(init?: Partial<CaptionScript>) {
  const captionScript = Object.assign(
    copy<CaptionScript>(initialCaptionScript),
    init
  );

  if (captionScript.opacity == null || isNaN(captionScript.opacity)) {
    captionScript.opacity = 100;
  }

  return captionScript;
}
