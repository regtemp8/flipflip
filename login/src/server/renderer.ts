import createCache, { type EmotionCache } from "@emotion/cache";

export function createEmotionCache(nonce?: string): EmotionCache {
  return createCache({ key: "css", nonce });
}
