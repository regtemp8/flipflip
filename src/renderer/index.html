<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <title>FlipFlip</title>
    <!-- Responsive meta tag -- ensures proper rendering and touch zooming for all devices -->
    <meta name="viewport" content="minimum-scale=1, initial-scale=1, width=device-width, shrink-to-fit=no" />
    <!-- Roboto Font -->
    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap" />
    <!-- Font Icons -->
    <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons" />

    <meta
      http-equiv="Content-Security-Policy"
      content="frame-src https://www.imagefap.com http://www.imagefap.com https://hypno.nimja.com/ http://hypno.nimja.com/; font-src 'self' https://fonts.gstatic.com; style-src 'self' https://fonts.googleapis.com 'unsafe-inline'"
    />
  </head>

  <body>
    <div id="app" class="app"></div>
  </body>
</html>
