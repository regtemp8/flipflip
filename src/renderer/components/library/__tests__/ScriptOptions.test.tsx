import React from "react";
import { describe, it, expect } from "@jest/globals";
import renderer from "react-test-renderer";
import ScriptOptions from "../ScriptOptions";
import TestProvider from "../../../../../test/util/TestProvider";
import CaptionScript from "../../../data/CaptionScript";

jest.mock('../FontOptions', () => 'FontOptions');

describe("ScriptOptions", () => {
  it("should match snapshot", () => {
    const script = new CaptionScript()
    const component = renderer.create(
      <TestProvider>
        <ScriptOptions
          script={script}
          onCancel={() => {}}
          onFinishEdit={(common) => {}}
        />
      </TestProvider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
