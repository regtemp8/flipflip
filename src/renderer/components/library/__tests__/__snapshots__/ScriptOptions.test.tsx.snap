// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScriptOptions should match snapshot 1`] = `
<div
  TransitionComponent={[Function]}
  className="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="presentation"
>
  <div
    data-testid="sentinelStart"
    onFocus={[Function]}
    tabIndex={0}
  />
  <div
    className="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    onMouseDown={[Function]}
  >
    <div
      aria-describedby="edit-description"
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <div
        className="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Edit script options
        </h6>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-ofrcno-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-ofrcno-MuiGrid-root"
            >
              <div
                className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <div
                  className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
                  style={
                    {
                      "minHeight": "0px",
                    }
                  }
                >
                  <div
                    className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                  >
                    <div
                      className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                    >
                      <label
                        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                      >
                        <span
                          className="MuiSwitch-root MuiSwitch-sizeSmall css-1ycxvqq-MuiSwitch-root"
                        >
                          <span
                            className="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-1flv81n-MuiButtonBase-root-MuiSwitch-switchBase"
                            onBlur={[Function]}
                            onContextMenu={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            tabIndex={null}
                          >
                            <input
                              checked={false}
                              className="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                              onChange={[Function]}
                              required={false}
                              type="checkbox"
                            />
                            <span
                              className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                            />
                          </span>
                          <span
                            className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                          />
                        </span>
                        <span
                          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                        >
                          Stop at End
                        </span>
                      </label>
                    </div>
                  </div>
                </div>
                <div
                  className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
                  style={
                    {
                      "minHeight": "0px",
                    }
                  }
                >
                  <div
                    className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                  >
                    <div
                      className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                    >
                      <label
                        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                      >
                        <span
                          className="MuiSwitch-root MuiSwitch-sizeSmall css-1ycxvqq-MuiSwitch-root"
                        >
                          <span
                            className="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-1flv81n-MuiButtonBase-root-MuiSwitch-switchBase"
                            onBlur={[Function]}
                            onContextMenu={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            tabIndex={null}
                          >
                            <input
                              checked={false}
                              className="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                              onChange={[Function]}
                              required={false}
                              type="checkbox"
                            />
                            <span
                              className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                            />
                          </span>
                          <span
                            className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                          />
                        </span>
                        <span
                          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                        >
                          Next Scene at End
                        </span>
                      </label>
                    </div>
                  </div>
                </div>
                <label
                  className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                >
                  <span
                    className="MuiSwitch-root MuiSwitch-sizeSmall css-1ycxvqq-MuiSwitch-root"
                  >
                    <span
                      className="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-1flv81n-MuiButtonBase-root-MuiSwitch-switchBase"
                      onBlur={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={null}
                    >
                      <input
                        checked={true}
                        className="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                        onChange={[Function]}
                        required={false}
                        type="checkbox"
                      />
                      <span
                        className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                      />
                    </span>
                    <span
                      className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                    />
                  </span>
                  <span
                    className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                  >
                    Sync Timestamp with Audio
                  </span>
                </label>
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              className="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
            >
              Script Opacity: 
              100
              %
            </div>
            <span
              className="MuiSlider-root MuiSlider-colorPrimary MuiSlider-sizeMedium css-19qy7iz-MuiSlider-root"
              defaultValue={100}
              onMouseDown={[Function]}
            >
              <span
                className="MuiSlider-rail css-14pt78w-MuiSlider-rail"
              />
              <span
                className="MuiSlider-track css-1gv0vcd-MuiSlider-track"
                style={
                  {
                    "left": "0%",
                    "width": "100%",
                  }
                }
              />
              <span
                className="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-hvku5w-MuiSlider-thumb"
                data-index={0}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                style={
                  {
                    "left": "100%",
                    "pointerEvents": undefined,
                  }
                }
              >
                <input
                  aria-labelledby="opacity-slider"
                  aria-orientation="horizontal"
                  aria-valuemax={100}
                  aria-valuemin={0}
                  aria-valuenow={100}
                  data-index={0}
                  disabled={false}
                  max={100}
                  min={0}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  step={1}
                  style={
                    {
                      "border": 0,
                      "clip": "rect(0 0 0 0)",
                      "direction": "ltr",
                      "height": "100%",
                      "margin": -1,
                      "overflow": "hidden",
                      "padding": 0,
                      "position": "absolute",
                      "whiteSpace": "nowrap",
                      "width": "100%",
                    }
                  }
                  type="range"
                  value={100}
                />
                <span
                  aria-hidden={true}
                  className="MuiSlider-valueLabel css-nnid7-MuiSlider-valueLabel"
                >
                  <span
                    className="MuiSlider-valueLabelCircle"
                  >
                    <span
                      className="MuiSlider-valueLabelLabel"
                    >
                      100%
                    </span>
                  </span>
                </span>
              </span>
            </span>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <FontOptions
              name="Blink"
              onUpdateOptions={[Function]}
              options={
                {
                  "border": false,
                  "borderColor": "#000000",
                  "borderpx": 5,
                  "color": "#FFFFFF",
                  "fontFamily": "Arial Black,Arial Bold,Gadget,sans-serif",
                  "fontSize": 20,
                }
              }
              systemFonts={[]}
            />
            <hr
              className="MuiDivider-root MuiDivider-fullWidth ScriptOptions-fontDivider-11 css-9mgopn-MuiDivider-root"
            />
            <FontOptions
              name="Caption"
              onUpdateOptions={[Function]}
              options={
                {
                  "border": false,
                  "borderColor": "#000000",
                  "borderpx": 3,
                  "color": "#FFFFFF",
                  "fontFamily": "Helvetica Neue,Helvetica,Arial,sans-serif",
                  "fontSize": 8,
                }
              }
              systemFonts={[]}
            />
            <hr
              className="MuiDivider-root MuiDivider-fullWidth ScriptOptions-fontDivider-11 css-9mgopn-MuiDivider-root"
            />
            <FontOptions
              name="Big Caption"
              onUpdateOptions={[Function]}
              options={
                {
                  "border": false,
                  "borderColor": "#000000",
                  "borderpx": 4,
                  "color": "#FFFFFF",
                  "fontFamily": "Arial Black,Arial Bold,Gadget,sans-serif",
                  "fontSize": 12,
                }
              }
              systemFonts={[]}
            />
            <hr
              className="MuiDivider-root MuiDivider-fullWidth ScriptOptions-fontDivider-11 css-9mgopn-MuiDivider-root"
            />
            <FontOptions
              name="Count"
              onUpdateOptions={[Function]}
              options={
                {
                  "border": false,
                  "borderColor": "#000000",
                  "borderpx": 5,
                  "color": "#FFFFFF",
                  "fontFamily": "Arial Black,Arial Bold,Gadget,sans-serif",
                  "fontSize": 20,
                }
              }
              systemFonts={[]}
            />
          </div>
        </div>
      </div>
      <div
        className="MuiDialogActions-root MuiDialogActions-spacing ScriptOptions-actions-5 css-hlj6pa-MuiDialogActions-root"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium css-1m4bfca-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Cancel
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-d0ukip-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Save
        </button>
      </div>
    </div>
  </div>
  <div
    data-testid="sentinelEnd"
    onFocus={[Function]}
    tabIndex={0}
  />
</div>
`;
